### PR TITLE
Dev 412 

### PR DESCRIPTION
**Pull request by kuuurt**

### CODE REVIEW

File: mediaire_toolbox/transaction_db/transaction_db.py [hunk 1]
```diff
@@ -125,28 +125,55 @@ def set_processing(self,
             self.session.rollback()
             raise
 
-    def set_failed(self, id_: int, cause: str):
+    def set_failed(self, id_: int, cause: str = None):
```
Comment (pereferrera) [inline mediaire_toolbox/transaction_db/transaction_db.py]: Why is it now possible that a transaction fails without a cause? I thought it's better to enforce always a cause

File: mediaire_toolbox/transaction_db/transaction_db.py [hunk 2]
```diff
@@ -125,28 +125,55 @@ def set_processing(self,
...
-            t.error = ''
+            if clear_error:
+                t.error = ''
+            self.session.commit()
+        except:
+            self.session.rollback()
+            raise
+
+    def set_skipped(self, id_: int, cause: str = None):
+        """to be called when the transaction is skipped.
```
Comment (pereferrera) [inline mediaire_toolbox/transaction_db/transaction_db.py]: I'm confused here. the doc says error field will be set to '' but we actually have an skipped cause error as parameter?

File: mediaire_toolbox/transaction_db/transaction_db.py [hunk 3]
```diff
@@ -125,28 +125,55 @@ def set_processing(self,
...
+                t.error = cause
             self.session.commit()
         except:
             self.session.rollback()
             raise
 
-    def set_completed(self, id_: int):
+    def set_completed(self, id_: int, clear_error: bool = True):
         """to be called when the transaction completes successfully.
         Error field will be set explicitly to '' and end_date automatically
```
Comment (pereferrera) [inline mediaire_toolbox/transaction_db/transaction_db.py]: only if clear_error = True

File: mediaire_toolbox/transaction_db/transaction_db.py (hunk 1 above)
Comment (kuuurt) [reply mediaire_toolbox/transaction_db/transaction_db.py]: Ah, before I used to set skipped as failed state. Will enfore cause.

**Pull request merged**